### PR TITLE
G-API: Implement line and circle draw operations

### PR DESCRIPTION
**Pull request by TolyaTalamanov**
```
build_gapi_standalone:Linux x64=ade-0.1.1d
build_gapi_standalone:Win64=ade-0.1.1d
build_gapi_standalone:Mac=ade-0.1.1d
build_gapi_standalone:Linux x64 Debug=ade-0.1.1d
```

### CODE REVIEW

**Comment by TolyaTalamanov** [comment]
@dmatveev

File: modules/gapi/include/opencv2/gapi/render.hpp [hunk 1]
```diff
@@ -50,12 +50,41 @@ struct Rect
 {
     cv::Rect   rect;  /** Coordinates of the rectangle < */
     cv::Scalar color; /** The rectangle color or brightness (grayscale image) < */
```
Comment (dmatveev) [inline modules/gapi/include/opencv2/gapi/render.hpp]: I believe this `< ` thing is not required actually? Please take a look how other public structure fields are documented in OpenCV. When I checked the generated documentation for OSD the last time, I saw that `< ` there in the generated comments.

File: modules/gapi/include/opencv2/gapi/render.hpp [hunk 2]
```diff
@@ -50,12 +50,41 @@ struct Rect
...
+    cv::Point  pt1;    /** The first point of the line segment  < */
+    cv::Point  pt2;    /** The second point of the line segment < */
+    cv::Scalar color;  /** The line color                       < */
+    int        thick;  /** The thickness of line                < */
+    int        lt;     /** The Type of the line. See #LineTypes < */
+    int        shift;  /** The number of fractional bits in the point coordinates */
+
+};
+
+using Prim  = util::variant<Text, Rect, Circle, Line>;
```
Comment (dmatveev) [inline modules/gapi/include/opencv2/gapi/render.hpp]: Let's make it
```
using Prim = util::variant
    < Text
    , Rect
    , Circle
    , Line
    >;
```
Here you are editing the same line again and again which makes harder to track changes in blame, in my case adding a primitive is a new line, and editing a primitive results in a single-line change (per primitive)

File: modules/gapi/src/api/render.cpp [hunk 3]
```diff
@@ -28,7 +28,21 @@ void cv::gapi::wip::draw::render(cv::Mat& bgr, const Prims& prims)
                 break;
             }
 
-            default: util::throw_error(std::logic_error("Unsupported draw event"));
+            case Prim::index_of<Circle>():
+            {
+                auto c_p = cv::util::get<Circle>(p);
```
Comment (dmatveev) [inline modules/gapi/src/api/render.cpp]: `const auto &`? (avoid possible extra copy?)

File: modules/gapi/include/opencv2/gapi/render.hpp (hunk 1 above)
Comment (TolyaTalamanov) [reply modules/gapi/include/opencv2/gapi/render.hpp]: Fixed

File: modules/gapi/include/opencv2/gapi/render.hpp (hunk 2 above)
Comment (TolyaTalamanov) [reply modules/gapi/include/opencv2/gapi/render.hpp]: Fixed

File: modules/gapi/src/api/render.cpp (hunk 3 above)
Comment (TolyaTalamanov) [reply modules/gapi/src/api/render.cpp]: Fixed

File: modules/gapi/include/opencv2/gapi/render.hpp (hunk 1 above)
Comment (dmatveev) [reply modules/gapi/include/opencv2/gapi/render.hpp]: image

Much nicer now, thanks!

**Pull request merged**